src/libOpenImageIO/maketexture.cpp

```diff
@@ -428,7 +428,110 @@ lightprobe_to_envlatl (ImageBuf &dst, const ImageBuf &src, bool y_is_up,
     return true;
 }
 
+// compute image gradient in s,t space using a Sobel filter
+static void 
+sobel_gradient(const ImageBuf &src, int x, int y, int width, int height, float *h, float* dh_ds, float *dh_dt ){
+            
+    ASSERT (src.spec().nchannels==1);
```

Comment:
Instead of asserting (and checking on every pixel), I think it would be better to detect the wrong number of input channels farther upstream and give a helpful error, and/or just use channel [0] regardless of how many are in the file.